### ISSUE TITLE
bad argument #1 to 'set' (expecting number or Tensor or Storage)

### ISSUE THREAD

**Issue by redasu** [cardwing/Codes-for-Lane-Detection#274]
When trying to train ENet-Label-Torch model I'm getting error like:
/home/redas/torch/install/bin/luajit: /home/redas/torch/install/share/lua/5.1/nn/Module.lua:286: bad argument #1 to 'set' (expecting number or Tensor or Storage)
stack traceback:
	[C]: in function 'set'
	/home/redas/torch/install/share/lua/5.1/nn/Module.lua:286: in function 'flatten'
	/home/redas/torch/install/share/lua/5.1/nn/Module.lua:326: in function 'getParameters'
	./train.lua:36: in function '__init'
	/home/redas/torch/install/share/lua/5.1/torch/init.lua:91: in function </home/redas/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'Trainer'
	main.lua:36: in main chunk
	[C]: in function 'dofile'
	...edas/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x56393138f570
Any suggestions how to fix?

Trying to run on a single rtx2080s. Ubuntu 18.04 CUDA 9.1 CDNN v4

**Comment by cardwing** [comment]
See [this issue](https://github.com/cardwing/Codes-for-Lane-Detection/issues/191).

**Comment by redasu** [comment]
@cardwing that issue does not seem to have solution as it's stated that another system was used. any other suggestions? I am using default CULane dataset without any modifications.

**Comment by cardwing** [comment]
Maybe you can reinstall [Torch](http://torch.ch/docs/getting-started.html).

**Comment by cardwing** [comment]
Besides, have you tried the original SCNN repo? Do you meet the same problem? If yes, then the problem may lie in the Torch.

**Comment by redasu** [comment]
will try SCNN. maybe you know what's the highest cuda version that people used to train the model?

**Comment by cardwing** [comment]
I used cuda 8.0 when trying ENet-Label_torch. You can install cuda 8.0 to check if that problem still exists.

**Comment by redasu** [comment]
tried installing cuda 8, but did not help. tried SCNN and it's working.

**Comment by cardwing** [comment]
That's weird. Can you run the testing script of ENet-Label-Torch? The error occurs in getting model parameters, i.e., `self.params, self.gradParams = model:getParameters()`. Hence, the testing script should not work as well.

**Comment by redasu** [comment]
well testing script works, but it needs batches set to 2. Tried to modify train script too,but nothing seemed to work.

**Comment by cardwing** [comment]
Quite weird. I have not met this problem before. Maybe you can just load the model and check its parameters in the training script (comment other lines). Then, uncomment each line and find out which line does not work. If this cannot work, then I cannot come up with other solutions.

**Comment by redasu** [comment]
Will try that a bit later and will let you know.

Also want to ask, why pretrained model gives such bad results when testing with my own photos?
For example I resize original photo to same size as CULane pictures and replace one of those with my resized one, but network does not detect any lanes (0 0 0 0) and images in /predicts/ show almost nothing:

![00000](https://user-images.githubusercontent.com/45488335/82114674-e739ba80-9766-11ea-92ee-7e7ae18f857d.jpg)

![00000_3_avg](https://user-images.githubusercontent.com/45488335/82114679-ec970500-9766-11ea-9dcf-1cff01a77ff3.png)
I guess problems are because of streched out/poor quality images, since other images from google are working fairly OK.

**Comment by cardwing** [comment]
Maybe you can try some domain adaptation methods or few-shot learning methods to finetune the model.
